gix-object/src/tree/ref_iter.rs

```diff
@@ -42,6 +45,70 @@ impl<'a> TreeRef<'a> {
             .map(|idx| self.entries[idx])
     }
 
+    /// Follow a sequence of `path` components starting from this instance, and look them up one by one until the last component
+    /// is looked up and its tree entry is returned.
+    ///
+    /// # Performance Notes
+    ///
+    /// Searching tree entries is currently done in sequence, which allows the search to be allocation free. It would be possible
+    /// to reuse a vector and use a binary search instead, which might be able to improve performance over all.
+    /// However, a benchmark should be created first to have some data and see which trade-off to choose here.
+    ///
+    pub fn lookup_entry<I, P>(
+        &self,
+        odb: impl crate::FindExt,
+        buffer: &'a mut Vec<u8>,
+        path: I,
+    ) -> Result<Option<tree::Entry>, Error>
+    where
+        I: IntoIterator<Item = P>,
+        P: PartialEq<BStr>,
+    {
+        let mut path = path.into_iter().peekable();
+        let mut entries = self.entries.clone();
```

Comment:
The trick should be to stick to what's done in `gix` and implement this on the `TreeRefIter`, something I thought I saw in an earlier version.

Reply:
The reason I implemented the method on `TreeRef` was that, in its current iteration, the `gix-blame` PR uses `odb.find_tree` which returns a `TreeRef`. Since it seems I could switch to `odb.find_tree_iter` in the `gix-blame` PR instead, I’ll move `lookup_entry` to `TreeRefIter` as you suggested.

Reply:
Great! This has the chance to be a small but measurable performance boost as well. I still remember comparing two versions of this - what it is now (in `gix`), and one where it would allocate a `Vec` and binary-search through it. And it turned out the allocation really kills performance.